proxy/src/outbound.rs

```diff
@@ -99,3 +128,34 @@ where
         Buffer::new(balance, self.bind.executor()).map_err(|_| {})
     }
 }
+
+pub enum Discovery<B> {
+    LocalSvc(discovery::Watch<BindProtocol<B>>),
+    External(Option<(SocketAddr, BindProtocol<B>)>),
+}
+
+impl<B> Discover for Discovery<B>
+where
+    B: tower_h2::Body + 'static,
+{
+    type Key = SocketAddr;
+    type Request = http::Request<B>;
+    type Response = bind::HttpResponse;
+    type Error = <bind::Service<B> as tower::Service>::Error;
+    type Service = bind::Service<B>;
+    type DiscoverError = ();
+
+    fn poll(&mut self) -> Poll<Change<Self::Key, Self::Service>, Self::DiscoverError> {
+        match *self {
+            Discovery::LocalSvc(ref mut w) => w.poll(),
+            Discovery::External(ref mut opt) => {
```

Comment:
A side effect here that an external service will get inserted into the balancer, and then probably never removed. It seems like once we have some circuit breaking, this would take care of itself when/if the service closes, and outbound never requests this address again...

Reply:
Please add your GitHub comment here as a TODO comment in the source code.